pkg/testing/fetcher_local.go

```diff
@@ -58,10 +58,15 @@ func (f *localFetcher) Fetch(_ context.Context, operatingSystem string, architec
 
 	mainBuildfmt := "elastic-agent-%s-%s"
 	if f.snapshotOnly && !ver.IsSnapshot() {
-		mainBuildfmt = "elastic-agent-%s-SNAPSHOT-%s"
+		if ver.Prerelease() == "" {
+			ver = semver.NewParsedSemVer(ver.Major(), ver.Minor(), ver.Patch(), ver.Prerelease()+"SNAPSHOT", ver.BuildMetadata())
+		} else {
+			ver = semver.NewParsedSemVer(ver.Major(), ver.Minor(), ver.Patch(), ver.Prerelease()+"-SNAPSHOT", ver.BuildMetadata())
```

Comment:
I though the `-` before SNAPSHOT was always necessary? What am I missing here?

Reply:
This is not simple string concatenation: we are setting the *prerelease* string of a semver version.
The String() function of the semver object will format the version as 
`<major>.<minor>.<patch>[-<prerelease>][+<metadata>]`
If we had no prerelease set in the original version, we just need to  set the prerelease to `'SNAPSHOT'` and the version will have the expected format
If we already had a prerelease string set (like `er.1` for example) then we set the prerelease string to the original prerelease string suffixed with `'-SNAPSHOT'` (in the `er.1` example we get `er.1-SNAPSHOT` as prerelease string)